src/script/descriptor.cpp

```diff
@@ -841,6 +841,38 @@ class DescriptorImpl : public Descriptor
         return true;
     }
 
+    bool IsWatchOnly(const SigningProvider& arg) const override
```

Comment:
In 64213213b3691c9d93de95f880f820a640cd3c35 "descriptors: add IsWatchOnly()"

I think this function can be vastly simplified by calling `GetPrivKey` on all the `PubkeyProvider`, and `IsWatchOnly` on all subdescriptors.

nit: I don't like this function name since whether the descriptor is watchonly depends on the contents of the provided parameter. It's not a property like a `Is*` name implies. Watchonly also is not a property applied to descriptors as private keys are not actually part of the descriptor. Perhaps something like `HavePrivateKeys` instead.